Template-only components fixes (#1707)
ember-learn/guides-source: guides/release/upgrading/current-edition/glimmer-components.md

## Template-Only Components

In Octane, template-only components have only an `hbs` file and no `JavaScript` file.
Behind the scenes, template-only components inherit from `'@glimmer/component'`.

They can be thought of as _functional_ components, in the sense that their
output (the rendered template) is a pure function of their inputs (their
arguments). The fact that they can't have state makes them much easier to reason

> about in general, and less prone to errors.

Author: The first sentence is definitely misleading, the second paragraph is not necessarily wrong _per se_, but the analogy it uses is IMO even more complicated than what it is trying to explain in the first place, so it doesn't really serve a purpose.

/>
```

> Since Octane template-only components shares a subset of the features available
> to `@glimmer/component`, they can be "upgraded" seamless to a Glimmer component
> by adding a JavaScript file alongside the template when the need arises.

Reviewer: ```suggestion
Since Octane, a template-only component shares a subset of features that are available
in `@glimmer/component`. Such component can be seamlessly "upgraded" to a Glimmer component,
when you add a JavaScript file alongside the template.
```
